src/NumericalAlgorithms/Spectral/SwshCollocation.cpp

```diff
@@ -24,7 +24,7 @@ Collocation<Representation>::Collocation(const size_t l_max) noexcept
   sharp_make_gauss_geom_info(
       l_max_ + 1, 2 * l_max_ + 1, 0.0,
       detail::ComplexDataView<Representation>::stride(),
-      detail::ComplexDataView<Representation>::stride() * (1 * l_max_ + 1),
+      detail::ComplexDataView<Representation>::stride() * (2 * l_max_ + 1),
```

Comment:
This was a bug, and a frustrating one. The only test I can think of that elegantly catches it is to check the transform, because the resulting view object is technically reasonably well-formed, just useless for transforms.

I'll see if I can find where this gets put on the libsharp internals and see if I can explicitly check it in the collocation test.

Reply:
Alright, but don't put too much time into it :) I'm assuming it's testing indirectly now :P

Reply:
Yeah, the transforms break catastrophically if this is wrong.

Reply:
:+1: I will mark this as resolved then :)